tests/interrupt_read_only_trx_test.py

```diff
@@ -0,0 +1,179 @@
+    specificExtraNodeosArgs[pnodes]+=" --eos-vm-oc-cache-size-mb "
+    specificExtraNodeosArgs[pnodes]+=" 1 " # set small so there is churn
+    specificExtraNodeosArgs[pnodes]+=" --read-only-threads "
+    specificExtraNodeosArgs[pnodes]+=str(args.read_only_threads)
+    if args.eos_vm_oc_enable:
+        if platform.system() != "Linux":
+            Print("OC not run on Linux. Skip the test")
+            exit(0) # Do not fail the test
+        specificExtraNodeosArgs[pnodes]+=" --eos-vm-oc-enable "
+        specificExtraNodeosArgs[pnodes]+=args.eos_vm_oc_enable
+    if args.wasm_runtime:
+        specificExtraNodeosArgs[pnodes]+=" --wasm-runtime "
+        specificExtraNodeosArgs[pnodes]+=args.wasm_runtime
+    extraNodeosArgs=" --http-max-response-time-ms 99000000 --disable-subjective-api-billing false "
+    if cluster.launch(pnodes=pnodes, totalNodes=total_nodes, topo=topo, delay=delay, activateIF=True,
+                      specificExtraNodeosArgs=specificExtraNodeosArgs, extraNodeosArgs=extraNodeosArgs ) is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    Print ("Wait for Cluster stabilization")
+    # wait for cluster to start producing blocks
+    if not cluster.waitOnClusterBlockNumSync(3):
+        errorExit("Cluster never stabilized")
+
+    producerNode = cluster.getNode()
+    apiNode = cluster.nodes[-1]
+
+    eosioCodeHash = getCodeHash(producerNode, "eosio.token")
+    # eosio.* should be using oc unless oc tierup disabled
+    Utils.Print(f"search: executing {eosioCodeHash} with eos vm oc")
+    found = producerNode.findInLog(f"executing {eosioCodeHash} with eos vm oc")
+    assert( found or (noOC and not found) )
+
+def deployTestContracts():
+    Utils.Print("Create payloadless account and deploy payloadless contract")
+    payloadlessAccount = Account(payloadlessAccountName)
+    payloadlessAccount.ownerPublicKey = EOSIO_ACCT_PUBLIC_DEFAULT_KEY
+    payloadlessAccount.activePublicKey = EOSIO_ACCT_PUBLIC_DEFAULT_KEY
+    cluster.createAccountAndVerify(payloadlessAccount, cluster.eosioAccount, buyRAM=100000)
+
+    payloadlessContractDir="unittests/test-contracts/payloadless"
+    payloadlessWasmFile="payloadless.wasm"
+    payloadlessAbiFile="payloadless.abi"
+    producerNode.publishContract(payloadlessAccount, payloadlessContractDir, payloadlessWasmFile, payloadlessAbiFile, waitForTransBlock=True)
+
+def sendTransaction(account, action, data, auth=[], opts=None):
+    trx = {
+       "actions": [{
+          "account": account,
+          "name": action,
+          "authorization": auth,
+          "data": data
+      }]
+    }
+    return apiNode.pushTransaction(trx, opts, silentErrors=True)
+
+def sendReadOnlyForeverPayloadless():
+    try:
+        sendTransaction('payloadless', action='doitforever', data={}, auth=[], opts='--read')
+    except Exception as e:
+        Print("Ignore Exception in sendReadOnlyForeverPayloadless: ", repr(e))
+
+def timeoutTest():
```

Comment:
Can this test be placed into `read_only_trx_test.py` so test setup code is not repeated?

Reply:
I considered it, but the setup is different and seemed like it would overly complicate `read_only_trx_test.py`.